test/models/apple/api_response_test.rb

```diff
@@ -52,29 +52,79 @@ class TestResponse
     end
   end
 
-  describe ".join_on_apple_episode_id" do
-    it "uses the request metadata to join a set of resources responging to `apple_episode_id`" do
-      row_results = [{request_metadata: {apple_episode_id: "1"}, api_response: {foo: "111"}},
-        {request_metadata: {apple_episode_id: "2"}, api_response: {foo: "222"}},
-        {request_metadata: {apple_episode_id: "3"}, api_response: {foo: "333"}}]
-      row_results = row_results.map(&:with_indifferent_access)
-
-      resources = [
+  describe "joins" do
+    let(:row_results) do
+      [
+        {"request_metadata" => {"apple_episode_id" => "1", "bar" => "111"}, "api_response" => {"foo" => "111"}},
+        {"request_metadata" => {"apple_episode_id" => "2", "bar" => "222"}, "api_response" => {"foo" => "222"}},
+        {"request_metadata" => {"apple_episode_id" => "3", "bar" => "333"}, "api_response" => {"foo" => "333"}}
+      ]
+    end
+
+    let(:resources) do
+      [
         OpenStruct.new(apple_episode_id: "1", bar: "111"),
         OpenStruct.new(apple_episode_id: "2", bar: "222"),
         OpenStruct.new(apple_episode_id: "3", bar: "333")
       ]
+    end
+
+    describe "many to one joins" do
+      it "joins on apple_episode_id" do
+        zipped = TestResponse.join_many_on("apple_episode_id", resources, row_results)
```

Comment:
Joins!

(And tests++... I actually started with the tests and worked backwards looking at this PR)